### ISSUE TITLE
Program leaks ~10 MB/s when running from RenderDoc when I use multiple command buffers and Vulkan API

### ISSUE THREAD

**Issue by Benualdo** [baldurk/renderdoc#3512]
## Description of Bug

Hi, I am writing a program using DirectX12 and Vulkan and recently introduced render jobs/multiple command lists/command buffers. 

I have no issue with the DirectX12 version and RenderDoc, but with the Vulkan version I have a **memory leak** as soon as I enable usage of multiple command buffers **only when starting the program from RenderDoc**.

## Steps to reproduce

I don't think I can reproduce this issue on a small sample given how verbose the Vulkan API is :) but my project is open-source and can be easily compiled using Visual Studio 2022 (or I can provide binaries if needed) : https://github.com/vimontgames/vgframework
(Program can be run with `attachDebugger=true` to attach debugger before init if needed).

Use of multiple command buffer is disabled by default but can be enabled by clicking on "Render Jobs" in the "Renderer" tab in the top-right corner:
![image](https://github.com/user-attachments/assets/0d48a732-d041-4cd6-acc7-44e294d6e50c)

#### Before enabling multiple command buffers:
![image](https://github.com/user-attachments/assets/c34ad5f5-f260-4ecf-aaa7-c50e1b5f13cf)

#### After enabling multiple command buffers
![image](https://github.com/user-attachments/assets/c0c471b0-b858-465c-9db0-a4cf71f24a92)
Memory usage increases once then it's stable.

### Running from RenderDoc

The behaviour is different when running from RenderDoc.
![{8D6258E0-4DE0-447A-B360-BA9DD2E3FAEB}](https://github.com/user-attachments/assets/17f54ef2-6c49-413f-8503-b9be0b1bcddd)

As long as I don't enable multiple command lists, the memory usage is stable.
![image](https://github.com/user-attachments/assets/f9af9d6f-58aa-4f11-9999-3272de41fb12)

But as soon as I use more than one command buffer for submit, the memory usage keeps increasing. 

After ~10 seconds:
![image](https://github.com/user-attachments/assets/dc881c08-2ba4-4cc5-bc42-7722a61b8140)

After ~1 minute:
![image](https://github.com/user-attachments/assets/fecab575-ec27-402c-860e-002371583157)

And so on until I'm out of memory.

I completely understand that the issue could be on my side (even if I don't experience that memory leak when not running from RenderDoc) or you don't have time to look at anything else than a small repro sample but as there's a possiblity the issue might be on RenderDoc's side I wanted to report it.

Thanks.

## Environment

* RenderDoc version: 1.36
* Operating System: Windows 11 24H2
* Graphics API: Vulkan 1.3

**Comment by Zorro666** [comment]
I have not been able to reproduce this problem.
I cloned the repository @ 71a15f9555a4 and built a Vulkan debug x64 build.
Ran the executable from RenderDoc v1.36 using
![image](https://github.com/user-attachments/assets/cbdafc17-d414-48a1-a96c-72d0c54020b8)

![image](https://github.com/user-attachments/assets/ab913cf0-5839-4ef5-b325-d5c59c98a0d9)

and the memory usage is stable with `Render Jobs` on/off

I am running on an nVidia GPU

**Comment by Benualdo** [comment]
I can still reproduce the issue locally at commit 71a15f9555a4 on my AMD Radeon 7950 XTX (Adrenaline driver 24.8.1). I don’t currently have access to a computer with an NVIDIA GPU, but I ran the same test on a Windows 11 Surface Pro with an Intel GPU and did not encounter this issue. This suggests the problem might be vendor-specific or related to the AMD driver.

**Comment by Zorro666** [comment]
Thanks I tried on an AMD machine and still not seeing any memory leaks.

AMD Radeon RX 6800S 32.0.12019.1028

Perhaps there is a newer driver available for your GPU?
I do not have any Vulkan layers enabled or any other Vulkan layers installed (just RenderDoc and the Vulkan SDK ones).
I used Vulkan SDK 1.3.296.0

I might have the wrong reproduction steps

**Comment by baldurk** [comment]
I don't think this is RenderDoc related because when running the program on its own from explorer I can reproduce the described leak, which goes away when render jobs are disabled. The memory also seems to be freed when disabling and re-enabling that option.

I also sometimes get some asserts about a ring buffer: `Cannot allocate 85.33 MB in upload ring buffer because there is only 71.45/768.00 MB available`, I'm not sure if that's related.

EDIT: This is on my NV system, I didn't test on AMD as I was able to see the leak without using RenderDoc.

**Comment by Benualdo** [comment]
Thanks for investigating, I will look at the issue when running the program from explorer if I repro. This might indeed be related to the assert you had e.g. if startup is faster than when running from Visual Studio. Thanks.
